pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java

```diff
@@ -57,7 +57,7 @@ public String getServerVersion() {
     public int getServerVersionNum() {
         if (serverVersionNum != 0)
             return serverVersionNum;
-        return Utils.parseServerVersionStr(serverVersion);
+        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
```

Comment:
When looking into the code of this class, it looks like quite inconsistent about synchronization. 

Half of the methods are not synchronized though they access non-volatile inner variables. 

It definitely does not relate to that pull request.